### PR TITLE
Fix random queue creation order.

### LINKED ISSUE #102: Queue creation order random
As per #100, queues should not be created concurrently, instead they should either follow config file order, or in order that will satisfy deadletter queue dependencies.
@adamw suggest problem is here: https://github.com/adamw/elasticmq/blob/release-0.13.2/server/src/main/scala/org/elasticmq/server/ElasticMQServer.scala#L63

Example failure:

conf:
```
queues {
    queue1 {
        deadLettersQueue {
            name = "queue1-deadletters"
            maxReceiveCount = 3 
        }
    }
    queue1-deadletters { }
}
```
log:
```
00:30:58.090 [main] INFO  org.elasticmq.server.Main$ - Starting ElasticMQ server (0.13.2) ...
00:30:58.378 [elasticmq-akka.actor.default-dispatcher-4] INFO  akka.event.slf4j.Slf4jLogger - Slf4jLogger started
00:30:58.382 [elasticmq-akka.actor.default-dispatcher-4] DEBUG akka.event.EventStream - logger log1-Slf4jLogger started
00:30:58.385 [elasticmq-akka.actor.default-dispatcher-4] DEBUG akka.event.EventStream - Default Loggers started
00:30:58.471 [elasticmq-akka.actor.default-dispatcher-3] DEBUG c.t.sslconfig.akka.AkkaSSLConfig - Initializing AkkaSSLConfig extension...
00:30:58.472 [elasticmq-akka.actor.default-dispatcher-3] DEBUG c.t.sslconfig.akka.AkkaSSLConfig - buildHostnameVerifier: created hostname verifier: com.typesafe.sslconfig.ssl.DefaultHostnameVerifier@6c61a903
00:30:59.003 [elasticmq-akka.actor.default-dispatcher-4] DEBUG akka.io.TcpListener - Successfully bound to /0:0:0:0:0:0:0:0:9324
00:30:59.007 [elasticmq-akka.actor.default-dispatcher-6] INFO  o.e.rest.sqs.TheSQSRestServerBuilder - Started SQS rest server, bind address 0.0.0.0:9324, visible server address * (depends on incoming request path) 
00:30:59.048 [elasticmq-akka.actor.default-dispatcher-6] DEBUG o.elasticmq.actor.QueueManagerActor - Cannot create queue, its dead letters queue doesnt exists: QueueData(queue1,MillisVisibilityTimeout(30000),PT0S,PT0S,2017-03-10T00:30:59.013Z,2017-03-10T00:30:59.013Z,Some(DeadLettersQueueData(queue1-deadletters,3)),None)
00:30:59.049 [elasticmq-akka.actor.default-dispatcher-6] INFO  o.elasticmq.actor.QueueManagerActor - Creating queue QueueData(queue1-deadletters,MillisVisibilityTimeout(30000),PT0S,PT0S,2017-03-10T00:30:59.039Z,2017-03-10T00:30:59.039Z,None,None)
```

### PR DESCRIPTION
**Pull request by jononu**
Synchronously create queues in a topologically sorted order based on
their dependency graph.

Should fix #102 

If you decide to go a different direction to solve the problem, that is fine by me.  No pressure to accept it.  Also, I'm a total Scala newbie, so feedback is welcome.

### CODE REVIEW

File: server/src/main/scala/org/elasticmq/server/ElasticMQServer.scala [hunk 1]
```diff
@@ -80,4 +83,40 @@ class ElasticMQServer(config: ElasticMQServerConfig) extends Logging {
       deadLettersQueue = cq.deadLettersQueue.map(dlq => DeadLettersQueueData(dlq.name, dlq.maxReceiveCount))
     )
   }
+
+  private def sortCreateQueues(cqs: List[config.CreateQueue]): List[config.CreateQueue] = {
```
Comment (adamw) [inline server/src/main/scala/org/elasticmq/server/ElasticMQServer.scala]: Looks nice! Two small things:

1. Maybe a comment on *why* we need to sort the queues would be helpful for future generations?
2. How about moving this method to `config` (so that `config` would have a `sortedQueues` method or such), or even moving it to a separate class and adding some basic tests (empty input list, one-element, not sorted)

**Comment by jononu** [comment]
Updated, let me know what you think.  I wasn't sure about extracting the config case classes.

**Comment by adamw** [comment]
Thanks! Released in 0.13.3
